update_keb_installation.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python3
+# workflow-helpers
+# Copyright(C) 2021 Kevin Postlethwait
+#
+# This program is free software: you can redistribute it and / or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Update Kebechet installation details in DB during webhook workflows."""
+
+import os
+from tempfile import TemporaryDirectory
+
+import yaml
+import logging
+from ogr.services.github import GithubService
+
+from thoth.storages import GraphDatabase
+from thoth.common import cwd, RuntimeEnvironment
+from thoth.python import Project
+
+from thoth.workflow_helpers import __service_version__
+
+_SLUG = os.getenv("KEBECHET_SLUG", None)
+if _SLUG == "null":
+    _SLUG = None
+
+db = GraphDatabase()
+db.connect()
+
+_LOGGER = logging.getLogger("thoth.update_kebechet_installation")
+_LOGGER.info("Thoth workflow-helpers task: update kebechet installation details v%s", __service_version__)
+
+
+def update_keb_installation():
+    """Load files and pass them to storages update function."""
+    if _SLUG is None:
+        _LOGGER.info("No slug present, continuing to next step in task.")
+        return
+
+    service = GithubService(
+        github_app_id=os.getenv("GITHUB_APP_ID"),
+        github_private_key_path=os.getenv("GITHUB_PRIVATE_KEY_PATH"),
+    )  # TODO: extend to use other services
+
+    project = service.get_project(namespace=_SLUG.split("/")[0], repo=_SLUG.split("/")[1])
+
+    try:
+        raw_thoth_config = project.get_file_content(".thoth.yaml")
+    except Exception:
```

Comment:
Might be a good idea to be more specific here with the error handling. For instance, there might be a networking hiccup in the cluster and the file content will not be obtained. In that case we mark the given repo as inactive even though it is active. In such a case, an exception should be raised marking the task as errored and possibly retry next time. This way we make sure we have correct data in the databae.

Reply:
Also, please log errors that are happening so we have them in monitoring services and are aware of what's going on in the deployment.

Reply:
I decided to remove the error handling and just let the workflow error. WDYT?

Reply:
Sounds good if the task is restarted (I haven't checked wf configuration for this).